circuit_knitting_toolbox/circuit_cutting/cutting_evaluation.py

```diff
@@ -13,13 +13,139 @@
 
 from __future__ import annotations
 
+import copy
+from typing import Any, NamedTuple
 from collections.abc import Sequence
+from itertools import chain
+from multiprocessing.pool import ThreadPool
 
+import numpy as np
 from qiskit.circuit import QuantumCircuit, ClassicalRegister
-from ..utils.observable_grouping import CommutingObservableGroup
+from qiskit.quantum_info import PauliList
+from qiskit.primitives import BaseSampler, Sampler as TerraSampler
+from qiskit_aer.primitives import Sampler as AerSampler
+from qiskit.result import QuasiDistribution
 
+from ..utils.observable_grouping import CommutingObservableGroup, ObservableCollection
+from ..utils.iteration import strict_zip
+from .qpd import (
+    QPDBasis,
+    SingleQubitQPDGate,
+    TwoQubitQPDGate,
+    generate_qpd_samples,
+    decompose_qpd_instructions,
+    WeightType,
+)
+from .cutting_decomposition import decompose_observables
 
-def append_measurement_circuit(
+
+class ExperimentResults(NamedTuple):
+    """The results of running each sampled subexperiment on the backend."""
+
+    quasi_dists: list[list[list[tuple[QuasiDistribution, int]]]]
+    coeffs: Sequence[tuple[float, WeightType]]
+
+
+def execute_experiments(
+    circuits: QuantumCircuit | dict[str | int, QuantumCircuit],
+    observables: PauliList | dict[str | int, PauliList],
+    num_samples: int,
+    samplers: BaseSampler | dict[str | int, BaseSampler],
+) -> ExperimentResults:
+    r"""
+    Generate the sampled circuits, append the observables, and run the sub-experiments.
+
+    Args:
+        circuits: The circuit(s) resulting from decomposing nonlocal gates
+        observables: The observable(s) corresponding to the circuit(s). If
+            a :class:`~qiskit.QuantumCircuit` is submitted for the ``circuits`` argument,
+            a :class:`~qiskit.quantum_info.PauliList` is expected; otherwise, a mapping
+            from partition label to subobservables is expected.
+        num_samples: The number of samples to draw from the quasiprobability distribution
+        samplers: Sampler(s) on which to run the sub-experiments.
+
+    Returns:
+        A tuple containing a 3D list of length-2 tuples holding the quasi-distributions
+        and QPD bit information for each sub-experiment as well as the coefficients corresponding
+        to each unique subexperiment's sampling frequency.
+
+    Raises:
+        ValueError: The number of requested samples must be positive.
+        ValueError: The types of ``circuits`` and ``observables`` arguments are incompatible.
+        ValueError: ``SingleQubitQPDGate``\ s are not supported in unseparable circuits.
```

Comment:
Off the top of my head, the logic gets more complex with single qubit gates, because you need to look whether they're part of a greater decomposition and do everything in sync. Let me look closer for specifics

Reply:
We will leave this error for 0.2.0. Let's see how everything falls out with wire cutting and 0.3.0 and decide whether this is still a question